src/mlpack/methods/random_forest/random_forest_impl.hpp

```diff
@@ -431,6 +431,7 @@ void RandomForest<
 {
   // Train each tree individually.
   trees.resize(numTrees); // This will fill the vector with untrained trees.
+  double avgGain = 0.0;
```

Comment:
I think we might want to use an OpenMP reduce or something like this.  As it's written, `avgGain` is shared across all threads in the parallel loop for training each tree, so there may be locking issues.

Reply:
Done !

Reply:
:+1: